src/tribler/core/components/libtorrent/download_manager/download_manager.py

```diff
@@ -362,17 +362,25 @@ def get_session(self, hops=0):
 
         return self.ltsessions[hops]
 
-    def set_proxy_settings(self, ltsession, ptype, server=None, auth=None):
+    def set_proxy_settings(self, ltsession, ptype, server: Optional[Tuple] = None, auth: Optional[List] = None):
         """
         Apply the proxy settings to a libtorrent session. This mechanism changed significantly in libtorrent 1.1.0.
         """
         settings = {}
         settings["proxy_type"] = ptype
         settings["proxy_hostnames"] = True
         settings["proxy_peer_connections"] = True
-        if server and server[0] and server[1]:
-            settings["proxy_hostname"] = server[0]
-            settings["proxy_port"] = int(server[1])
+        if server:
+            proxy_port = None
+
+            try:
+                proxy_port = int(server[1])
```

Comment:
Under the possibility of `server[1]` being None which causes `int(server[1])` to raise `TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'`, I would suggest to keep the initial if statement:
```python3
if server and server[0] and server[1]:
```

Reply:
You are right about the technical possibility of server[1] being None. This could be due to wrongly formatted settings. If we start considering these corner cases, then we should also check that the value of server has exactly two items, as it is technically possible to form an incorrect parameter. For example:

```python
    fake_dlmgr.set_proxy_settings(session, 3, (None,))
```

To address these corner cases, I have to rewrite the function completely. Additional tests have also been added to highlight and test this possibility.